bitwarden_license/src/Commercial.Core/AdminConsole/Services/ProviderService.cs

```diff
@@ -393,6 +397,94 @@ public async Task AddOrganizationsToReseller(Guid providerId, IEnumerable<Guid>
         await _eventService.LogProviderOrganizationEventsAsync(insertedProviderOrganizations.Select(ipo => (ipo, EventType.ProviderOrganization_Added, (DateTime?)null)));
     }
 
+    private async Task UpdatePlanForExistingProviderAsync(IEnumerable<Guid> organizationIds, Provider provider)
+    {
+        // Check if provider was created on or after November 6, 2023; if so, do not update the plan to the 2020 plan.
+        if (provider.CreationDate >= new DateTime(2023, 11, 6))
+        {
+            return;
+        }
+
+        foreach (var orgId in organizationIds)
+        {
+            var organization = await _organizationRepository.GetByIdAsync(orgId);
+            var subscriptionItem = await GetSubscriptionItemAsync(organization.GatewaySubscriptionId, GetPlanId(organization.PlanType));
+            var newPlanType = GetPlanTypeFromPlan(organization.Plan, organization);
+            if (subscriptionItem != null)
+            {
+                await UpdateSubscriptionAsync(organization.GatewaySubscriptionId, subscriptionItem, GetPlanId(newPlanType));
+            }
+
+            await _organizationRepository.UpsertAsync(organization);
+        }
+    }
+
+    private async Task<Stripe.SubscriptionItem> GetSubscriptionItemAsync(string subscriptionId, string oldPlanId)
+    {
+        var subscrptionInfo = await _stripeAdapter.SubscriptionGetAsync(subscriptionId);
+        return subscrptionInfo.Items.Data.FirstOrDefault(item => item.Price.Id == oldPlanId);
+    }
+
+    private static string GetPlanId(PlanType planType)
+    {
+        return StaticStore.GetPlan(planType).PasswordManager.StripeSeatPlanId;
+    }
+
+    private async Task UpdateSubscriptionAsync(string subscriptionId, Stripe.SubscriptionItem subscriptionItem, string newPlanId)
+    {
+        try
+        {
+            if (subscriptionItem.Plan.Id != newPlanId)
+            {
+                await _stripeAdapter.SubscriptionUpdateAsync(subscriptionId,
+                    new Stripe.SubscriptionUpdateOptions
+                    {
+                        Items = new List<Stripe.SubscriptionItemOptions>
+                        {
+                            new Stripe.SubscriptionItemOptions { Id = subscriptionItem.Id, Price = newPlanId },
+                        }
+                    });
+            }
+        }
+        catch (Exception e)
+        {
+            throw new Exception("Unable to update existing plan on stripe");
+        }
+
+    }
+
+    private static PlanType GetPlanTypeFromPlan(string plan, Organization organization)
```

Comment:
I'm confused by what this method is for. It seems like it takes and organization plan string (the display name) and maps it back to the enum value - but the Organization object has a PlanType value on it already.